genai/example_test.go

```diff
@@ -53,7 +53,7 @@ func ExampleGenerativeModel_GenerateContent_config() {
 	defer client.Close()
 
 	model := client.GenerativeModel("gemini-pro")
-	model.Temperature = 0.9
+	model.SetTemperature(0.9)
```

Comment:
It's surprising that temprerature is the only one with the setter. Feels like leaking implementation details. Can we handle this on a lower layer when we translate to proto?

Reply:
If we keep Temperature a float32, then there's no good way to deal with it. We could handle it at a lower level by saying if Temperature is out of range (less than 0 or greater than 1), then use the default. That is basically what we do for the other config fields where we removed the pointer.

But then the default value in Go is not the default for the API. If you don't explicitly set Temperature to -1, you will get a temperature of 0. That is not a good experience.

The only ways to avoid this special case are:
- Restore all the config fields to pointers. (And possibly provide convenience Set methods for each, or better, provide a single function `Ptr[T](T) *T` that made it easy to make a pointer to a value.)

- Add some sort of Optional[T] type. We did something like this in the [storage client](https://pkg.go.dev/cloud.google.com/go/storage#BucketAttrsToUpdate), before we had generics. An `optional.Bool` is just an `any`, so the user can put `true` or `false` for the field, or omit it to get the default. That is not type-safe, though. If we used an Optional[T], the user would have to write something like `Opt(0.0)` which is basically as much work as writing `Ptr(0.0)`. So I'd prefer the former.

Reply:
OK, will go with `Ptr` for all the pointer fields in GenerationConfig.
Yes, same issue in Vertex. Will make the change there too.

Reply:
Done. It is more cumbersome than we thought because none of the proto types are the default Go types for untyped constants, so you have to write `Ptr[int32](0)` or `Ptr(int32(0))`.